rllib/utils/lambda_defaultdict.py

```diff
@@ -0,0 +1,45 @@
+from collections import defaultdict
+from typing import Any, Callable
+
+
+class LambdaDefaultDict(defaultdict):
+    """A defaultdict subclass that takes a lambda function as a default_factory.
```

Comment:
What is the diff between this and the normal defaultdict? I feel like I am not clear so much on the diff.

Reply:
It's described one line below:
```
The lambda function is expected to accept a single argument: the missing key.

If a missing key is accessed, the provided lambda function is called with the
missing key as its argument. The returned value is stored in the dictionary
under that key and returned.
```
I'll add the description of why this is different (and important) wrt the standard defaultdict.

Reply:
Fixed description. Made it more clear what the difference is.

Reply:
oh independence of the default value based on key is the important key point. 👍